src/utils/HTTPTransport.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from 'chai';
+import Sinon, { SinonFakeXMLHttpRequest, SinonFakeXMLHttpRequestStatic } from 'sinon';
+import HTTPTransport from './HTTPTransport';
+
+describe('Тест HTTPTransport', () => {
+  let xhr: SinonFakeXMLHttpRequestStatic;
+  let instance: HTTPTransport;
+  const requests: SinonFakeXMLHttpRequest[] = [];
+
+  beforeEach(() => {
+    xhr = Sinon.useFakeXMLHttpRequest();
+
+    // @ts-ignore
+    global.XMLHttpRequest = xhr;
+
+    xhr.onCreate = ((request: SinonFakeXMLHttpRequest) => {
+      requests.push(request);
+    });
+
+    instance = new HTTPTransport('/auth');
+  });
+
+  afterEach(() => {
+    requests.length = 0;
```

Comment:
лучше сбрасывать значение расшаренной переменной присваивая новое значение, не меняя его свойства.
```suggestion
    requests = []
```

1) так проще читается
2) соблюдается принцип иммутабельности данных
3) нужно будет изменить const на let на 8 строке